### PR TITLE
Updates the databuilder directory name after deprecating databuilder submodule

### PR DESCRIPTION
**Pull request by marcin-mazurek**
### Summary of Changes

docs: update databuilder directory name after deprecating databuilder submodule

### Tests

No tests - just docs change

### Documentation

The installation manual required an update following the deprecation of https://github.com/amundsen-io/amundsendatabuilder repository

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [X] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [X] PR includes a summary of changes.
- [X] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [X] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

**Review by feng-tao — APPROVED** [review]

**Comment by mgorsk1** [comment]
thanks for pr @marcin-mazurek ! does it cover all obsolete databuilder references?

**Comment by feng-tao** [comment]
+1 on @mgorsk1, I wonder whether it covers all the note for amundendatabuilder repo.

File: docs/installation.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ The following instructions are for setting up a version of Amundsen using Docker
     If it's your first time, you may want to proactively go through [troubleshooting](troubleshooting) steps, especially the first one related to heap memory for ElasticSearch and Docker engine memory allocation (leading to Docker error 137).
 4. Ingest provided sample data into Neo4j by doing the following: _(Please skip if you are using Atlas backend)_
 
-   * In a separate terminal window, change directory to the [amundsendatabuilder](https://github.com/amundsen-io/amundsendatabuilder) submodule.
+   * In a separate terminal window, change directory to [databuilder](https://github.com/amundsen-io/amundsen/tree/main/databuilder).
```
Comment (mgorsk1) [inline docs/installation.md:22]: does it cover all obsolete entries for databuilder repo?

Comment (Fokko) [reply docs/installation.md:22]: Looks like there are many more:
```
fokkodriesprong@MacBook-Pro-van-Fokko amundsen % grep -R "amundsendatabuilder" . | wc -l
     337
```

**Review by Fokko — APPROVED** [review]

File: docs/installation.md (hunk 1 above)
Comment (dorianj) [reply docs/installation.md:22]: Looks like about 1/2 of those are from CHANGELOGs and are correct, but the rest do need to be fixed

**Pull request merged**